extensions/amp-story/1.0/test/test-amp-story.js

```diff
@@ -1198,6 +1199,147 @@ describes.realWin(
         });
       });
 
+      describe('amp-story-subscriptions navigation', () => {
+        let subscriptionsEl;
+
+        beforeEach(async () => {
+          toggleExperiment(win, 'amp-story-subscriptions', true);
+
+          await createStoryWithPages(4, [
+            'cover',
+            'page-1',
+            'page-2',
+            'page-3',
+          ]);
+          subscriptionsEl = win.document.createElement(
+            'amp-story-subscriptions'
+          );
+          story.element.appendChild(subscriptionsEl);
+
+          // buildCallback() could be omitted as long as createStoryWithPages()
+          // is called because the getImpl() call inside it.
+          await story.layoutCallback();
+        });
+
+        it('should continue navigating to locked pages after the subscription state gets resolved to granted', async () => {
+          // Block access with pending subscriptions state.
+          await story.switchTo_('page-1');
+          story.switchTo_('page-2');
+          expect(story.activePage_.element.id).to.equal('page-1');
+
+          // Once grant status resolves to true, goes back to normal story experiences.
+          story.storeService_.dispatch(
+            Action.TOGGLE_SUBSCRIPTIONS_STATE,
+            SubscriptionsState.GRANTED
+          );
+          await nextTick();
+          expect(story.activePage_.element.id).to.equal('page-2');
+
+          await story.switchTo_('page-3');
+          expect(story.activePage_.element.id).to.equal('page-3');
+        });
+
+        it('should block and switch dialog UI state after the subscription state gets resolved to blocked', async () => {
+          // Block access with pending subscriptions state.
+          await story.switchTo_('page-1');
+          story.switchTo_('page-2');
+          expect(story.activePage_.element.id).to.equal('page-1');
+
+          // Once grant status resolves to false, blocks the access to all locked pages and show the paywall.
+          // Note: would still go to the paywall page, e.g. page-2.
+          const dispatchSpy = env.sandbox.spy(story.storeService_, 'dispatch');
+          story.storeService_.dispatch(
+            Action.TOGGLE_SUBSCRIPTIONS_STATE,
+            SubscriptionsState.BLOCKED
+          );
+          await nextTick();
+          expect(story.activePage_.element.id).to.equal('page-2');
+
+          await setTimeout(() => {
```

Comment:
you can force the clock to advance instead of setting another timeout, this makes tests very slow

Reply:
What do you mean by forcing clock to advance? The timeout here is to mirror the 2 seconds delay to show the paywall. Is there a better way to set timeout?

Reply:
Something like this https://github.com/ampproject/amphtml/blob/0586b0a55be6ea7a6716736734fd19d7701c58ce/extensions/amp-access/0.1/test/test-amp-access.js#L265

This allows us to fake 2 second time increase without actually waiting 2 seconds.